# **MÓDULO 14 - Exercício**
# Pré Processamento dos Dados - A primeira etapa Pré Modelagem

Para nossa atividade usaremos uma base de churn também, porém essa base não diz respeito a serviços bancários e sim serviços de internet, telefone e tv - Telecomunicação.

Vamos realizar aqui as etapas de limpeza e tratamento de dados vistos durante a aula, porém aplicados a essa base nova.

In [1]:
import seaborn as sns
import matplotlib.pyplot as plt
import pandas as pd

In [587]:
#Lembrem-se sempre de alterar a importação dos dados de acordo com o diretório de vocês.
df = pd.read_csv("C:/Users/vinicius/Downloads/CHURN_TELECON_MOD08_TAREFA.csv", delimiter=';')
df.head(10)

customerID  Genero  Idoso Casado Dependents  Tempo_como_Cliente  \
0  7590-VHVEG  Female      0    Yes         No                   1   
1  5575-GNVDE    Male      0     No         No                  34   
2  3668-QPYBK    Male      0     No         No                   2   
3  7795-CFOCW    Male      0     No         No                  45   
4  9237-HQITU     NaN      0     No         No                   2   
5  9305-CDSKC     NaN      0     No         No                   8   
6  1452-KIOVK     NaN      0     No        Yes                  22   
7  6713-OKOMC     NaN      0     No         No                  10   
8  7892-POOKP     NaN      0    Yes         No                  28   
9  6388-TABGU     NaN      0     No        Yes                  62   

  PhoneService Servico_Internet Servico_Seguranca Suporte_Tecnico StreamingTV  \
0           No              DSL                No              No          No   
1          Yes              DSL               Yes              No          No   
2          Yes              DSL               Yes              No          No   
3           No              DSL               Yes             Yes          No   
4          Yes      Fiber optic                No              No          No   
5          Yes      Fiber optic                No              No         Yes   
6          Yes      Fiber optic                No              No         Yes   
7           No              DSL               Yes              No          No   
8          Yes      Fiber optic                No             Yes         Yes   
9          Yes              dsl               Yes              No          No   

    Tipo_Contrato              PaymentMethod  Pagamento_Mensal  Total_Pago  \
0  Month-to-month           Electronic check             29.85       29.85   
1        One year               Mailed check             56.95     1889.50   
2  Month-to-month               Mailed check             53.85      108.15   
3        One year  Bank transfer (automatic)               NaN     1840.75   
4  Month-to-month           Electronic check               NaN      151.65   
5  Month-to-month           Electronic check               NaN      820.50   
6  Month-to-month    Credit card (automatic)               NaN     1949.40   
7  Month-to-month               Mailed check               NaN      301.90   
8  Month-to-month           Electronic check               NaN     3046.05   
9        One year  Bank transfer (automatic)               NaN     3487.95   

  Churn  
0    No  
1    No  
2   Yes  
3    No  
4   Yes  
5   NaN  
6   NaN  
7   NaN  
8   NaN  
9   NaN

Legenda dos dados:

*   **CustomerID** : Esse é o ID único de cada cliente nosso da rede de produtos de telecomunicação.

*   **Genero** : O genero dos nossos clientes.

*   **Idoso** : Se nosso cliente tem acima de 60 anos ou não. (Sim = 1 | Não = 0)

*   **Casado** : Se nosso cliente é casado ou não.

*   **Dependents** : Se cliente tem dependentes.

*   **Tempo_como_Cliente** : Tempo em meses que nosso cliente está com a nossa empresa.

*   **PhoneService** : Se ele contratou serviço telefônico com a nossa empresa.

*   **Servico_Internet** : Qual o tipo de serviço de internet contratado.

*   **Servico_Seguranca** : Se ele contratou serviço de segurança com a nossa empresa.

*   **Suporte_Tecnico** : Se já acionou o suporte técnico.

*   **StreamingTV** : Se ele contratou serviço de streaming para TV.

*   **Tipo_Contrato** : Qual o tipo de contrato dos nossos clientes, se era mensal, anual etc.

*   **Tipo_Contrato** : Qual o tipo de contrato dos nossos clientes, se era mensal, anual etc.

*   **PaymentMethod** : Forma de Pagamento.

*   **Pagamento_Mensal** : Quanto o cliente pagava mensalmente.

*   **Total_Pago** : Total já pago pelo cliente para nossa companhia.

*   **Churn** : Nossa coluna mais importante, se nosso cliente nos abandonou ou não.




# 1 - Carregue a base, verifique se os tipos de dados estão de acordo com cada coluna e caso não estejam realize a transformação dos tipos de dados.

In [591]:
#seu código aqui
df.dtypes

customerID             object
Genero                 object
Idoso                   int64
Casado                 object
Dependents             object
Tempo_como_Cliente      int64
PhoneService           object
Servico_Internet       object
Servico_Seguranca      object
Suporte_Tecnico        object
StreamingTV            object
Tipo_Contrato          object
PaymentMethod          object
Pagamento_Mensal      float64
Total_Pago            float64
Churn                  object
dtype: object

In [593]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2500 entries, 0 to 2499
Data columns (total 16 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   customerID          2500 non-null   object 
 1   Genero              2488 non-null   object 
 2   Idoso               2500 non-null   int64  
 3   Casado              2500 non-null   object 
 4   Dependents          2500 non-null   object 
 5   Tempo_como_Cliente  2500 non-null   int64  
 6   PhoneService        1018 non-null   object 
 7   Servico_Internet    2500 non-null   object 
 8   Servico_Seguranca   2500 non-null   object 
 9   Suporte_Tecnico     2500 non-null   object 
 10  StreamingTV         2500 non-null   object 
 11  Tipo_Contrato       2500 non-null   object 
 12  PaymentMethod       2500 non-null   object 
 13  Pagamento_Mensal    2175 non-null   float64
 14  Total_Pago          2500 non-null   float64
 15  Churn               2495 non-null   object 
dtypes: flo

In [595]:
#seu código aqui
df['Genero'] = df['Genero'].fillna('Unknown')

# Converter colunas categóricas
categoricas = ['Genero', 'Casado', 'Dependents', 'Servico_Internet', 
               'Servico_Seguranca', 'Suporte_Tecnico', 'StreamingTV', 'Tipo_Contrato', 
               'PaymentMethod', 'Churn']
for coluna in categoricas:
    df[coluna] = df[coluna].astype('category')

# Converter 'Idoso' para int64 (substituir NaN por 0 se houver)
df['Idoso'] = df['Idoso'].fillna(0).astype('int64')

# Converter 'Tempo_como_Cliente' para int64 (substituir NaN por 0 se houver)
df['Tempo_como_Cliente'] = df['Tempo_como_Cliente'].fillna(0).astype('int64')

# Converter 'Pagamento_Mensal' e 'Total_Pago' para float64, tratando valores ausentes
df['Pagamento_Mensal'] = pd.to_numeric(df['Pagamento_Mensal'], errors='coerce')
df['Total_Pago'] = pd.to_numeric(df['Total_Pago'], errors='coerce')

# Verificar os novos tipos de dados
print("Novos tipos de dados após transformação:")
df.dtypes
print("\n")

# Exibir as primeiras linhas para confirmar
print("Primeiras linhas após transformação:")
df.head()

Novos tipos de dados após transformação:


Primeiras linhas após transformação:


customerID   Genero  Idoso Casado Dependents  Tempo_como_Cliente  \
0  7590-VHVEG   Female      0    Yes         No                   1   
1  5575-GNVDE     Male      0     No         No                  34   
2  3668-QPYBK     Male      0     No         No                   2   
3  7795-CFOCW     Male      0     No         No                  45   
4  9237-HQITU  Unknown      0     No         No                   2   

  PhoneService Servico_Internet Servico_Seguranca Suporte_Tecnico StreamingTV  \
0           No              DSL                No              No          No   
1          Yes              DSL               Yes              No          No   
2          Yes              DSL               Yes              No          No   
3           No              DSL               Yes             Yes          No   
4          Yes      Fiber optic                No              No          No   

    Tipo_Contrato              PaymentMethod  Pagamento_Mensal  Total_Pago  \
0  Month-to-month           Electronic check             29.85       29.85   
1        One year               Mailed check             56.95     1889.50   
2  Month-to-month               Mailed check             53.85      108.15   
3        One year  Bank transfer (automatic)               NaN     1840.75   
4  Month-to-month           Electronic check               NaN      151.65   

  Churn  
0    No  
1    No  
2   Yes  
3    No  
4   Yes

In [597]:
df.dtypes

customerID              object
Genero                category
Idoso                    int64
Casado                category
Dependents            category
Tempo_como_Cliente       int64
PhoneService            object
Servico_Internet      category
Servico_Seguranca     category
Suporte_Tecnico       category
StreamingTV           category
Tipo_Contrato         category
PaymentMethod         category
Pagamento_Mensal       float64
Total_Pago             float64
Churn                 category
dtype: object

Explicação das Transformações
Colunas Categóricas: Usamos category para reduzir o uso de memória e facilitar análises estatísticas (ex.: contagens, proporções).

Colunas Numéricas:
Idoso e Tempo_como_Cliente foram mantidos ou convertidos para int64, pois representam contagens discretas.
Pagamento_Mensal e Total_Pago foram convertidos para float64 para lidar com valores decimais (ex.: 29.85) e valores ausentes (convertidos para NaN).

Valores Ausentes:
Genero foi preenchido com "Unknown" para manter consistência.
Pagamento_Mensal e Total_Pago usam errors='coerce' para converter strings inválidas (como vazios) para NaN.

# 2 - Esse exercício faremos por etapas:
A) Verifique se temos colunas com dados faltantes e traga a % dos dados faltantes por coluna.

In [601]:
df.isnull().values.any()

True

In [603]:
df.isnull().sum()

customerID               0
Genero                   0
Idoso                    0
Casado                   0
Dependents               0
Tempo_como_Cliente       0
PhoneService          1482
Servico_Internet         0
Servico_Seguranca        0
Suporte_Tecnico          0
StreamingTV              0
Tipo_Contrato            0
PaymentMethod            0
Pagamento_Mensal       325
Total_Pago               0
Churn                    5
dtype: int64

In [605]:
(df.isnull().sum() / len(df)) * 100

customerID             0.00
Genero                 0.00
Idoso                  0.00
Casado                 0.00
Dependents             0.00
Tempo_como_Cliente     0.00
PhoneService          59.28
Servico_Internet       0.00
Servico_Seguranca      0.00
Suporte_Tecnico        0.00
StreamingTV            0.00
Tipo_Contrato          0.00
PaymentMethod          0.00
Pagamento_Mensal      13.00
Total_Pago             0.00
Churn                  0.20
dtype: float64

In [607]:
# Tratar strings vazias como NaN
df = df.replace(r'^\s*$', pd.NA, regex=True)

# Verificar valores únicos em PhoneService antes do tratamento
print("Valores únicos em PhoneService antes do tratamento:")
print(df['PhoneService'].unique())
print("\n")

# Verificar a distribuição de valores não nulos para decidir a substituição
print("Distribuição de valores em PhoneService (antes de tratar nulos):")
print(df['PhoneService'].value_counts(dropna=False))
print("\n")

# Substituir valores ausentes em PhoneService pela moda
phone_service_mode = df['PhoneService'].mode()
if not phone_service_mode.empty:
    moda_valor = phone_service_mode[0]
    print(f"Moda de PhoneService: {moda_valor}")
    df['PhoneService'] = df['PhoneService'].fillna(moda_valor)
else:
    print("Moda de PhoneService não encontrada, usando 'No' como padrão.")
    df['PhoneService'] = df['PhoneService'].fillna('No')

# Verificar valores únicos após preenchimento
print("Valores únicos em PhoneService após preenchimento:")
print(df['PhoneService'].unique())
print("\n")

# Converter PhoneService para float (Yes -> 0.1, No -> 0.0)
df['PhoneService'] = df['PhoneService'].str.lower().replace({'yes': 0.1, 'no': 0.0}).astype('float')

# Verificar valores únicos após conversão
print("Valores únicos em PhoneService após conversão para float:")
print(df['PhoneService'].unique())
print("\n")

# Verificar o tipo de dado após conversão
print("Tipos de dados após conversão:")
print(df.dtypes)
print("\n")

# Exibir as primeiras linhas para confirmar
print("Primeiras linhas após transformação:")
df[['PhoneService']].head()

Valores únicos em PhoneService antes do tratamento:
['No' 'Yes' nan]


Distribuição de valores em PhoneService (antes de tratar nulos):
PhoneService
NaN    1482
Yes     931
No       87
Name: count, dtype: int64


Moda de PhoneService: Yes
Valores únicos em PhoneService após preenchimento:
['No' 'Yes']


Valores únicos em PhoneService após conversão para float:
[0.  0.1]


Tipos de dados após conversão:
customerID              object
Genero                category
Idoso                    int64
Casado                category
Dependents            category
Tempo_como_Cliente       int64
PhoneService           float64
Servico_Internet      category
Servico_Seguranca     category
Suporte_Tecnico       category
StreamingTV           category
Tipo_Contrato         category
PaymentMethod         category
Pagamento_Mensal       float64
Total_Pago             float64
Churn                 category
dtype: object


Primeiras linhas após transformação:


C:\Users\vinicius\AppData\Local\Temp\ipykernel_15152\116155782.py:30: FutureWarning: Downcasting behavior in `replace` is deprecated and will be removed in a future version. To retain the old behavior, explicitly call `result.infer_objects(copy=False)`. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df['PhoneService'] = df['PhoneService'].str.lower().replace({'yes': 0.1, 'no': 0.0}).astype('float')


PhoneService
0           0.0
1           0.1
2           0.1
3           0.0
4           0.1

In [609]:
(df.isnull().sum() / len(df)) * 100

customerID             0.0
Genero                 0.0
Idoso                  0.0
Casado                 0.0
Dependents             0.0
Tempo_como_Cliente     0.0
PhoneService           0.0
Servico_Internet       0.0
Servico_Seguranca      0.0
Suporte_Tecnico        0.0
StreamingTV            0.0
Tipo_Contrato          0.0
PaymentMethod          0.0
Pagamento_Mensal      13.0
Total_Pago             0.0
Churn                  0.2
dtype: float64

B) Exclua nessa etapa as linhas das colunas com dados faltantes que você analisar que esse método é o ideal. Justifique o motivo de estar excluindo essas linhas.

In [612]:
# Verificar o número de linhas antes da exclusão
print(f"Número de linhas antes da exclusão: {len(df)}")

# Excluir linhas com valores ausentes em Pagamento_Mensal e Churn
colunas_com_faltantes = ['Pagamento_Mensal', 'Churn']
df = df.dropna(subset=colunas_com_faltantes)

# Verificar o número de linhas após a exclusão
print(f"Número de linhas após a exclusão: {len(df)}")
print("\n")

# Verificar percentual de dados faltantes após exclusão
percentual_faltante = (df.isna().sum() / len(df)) * 100
print("Porcentagem de dados faltantes por coluna após exclusão (%):")
print(percentual_faltante.round(2))

Número de linhas antes da exclusão: 2500
Número de linhas após a exclusão: 2175


Porcentagem de dados faltantes por coluna após exclusão (%):
customerID            0.0
Genero                0.0
Idoso                 0.0
Casado                0.0
Dependents            0.0
Tempo_como_Cliente    0.0
PhoneService          0.0
Servico_Internet      0.0
Servico_Seguranca     0.0
Suporte_Tecnico       0.0
StreamingTV           0.0
Tipo_Contrato         0.0
PaymentMethod         0.0
Pagamento_Mensal      0.0
Total_Pago            0.0
Churn                 0.0
dtype: float64


Impacto da Exclusão: A perda de cerca de 317 linhas (4,5% do total) é aceitável para este dataset, especialmente porque garante dados completos para análises futuras.

C) Nessa etapa substitua pela média, mediana ou moda as linhas das colunas com valores faltantes que você analisou e entendeu que esse era o melhor método. Não se esqueça de justificar sua escolha.

**Dica variáveis numéricas:** Não se esqueça de verificar a distrbuição dos dados, média e mediana.


In [617]:
# Tratar strings vazias como NaN
df = df.replace(r'^\s*$', pd.NA, regex=True)

# Substituir valores ausentes
df['PhoneService'] = df['PhoneService'].fillna(df['PhoneService'].mode()[0])  # Moda
df['Pagamento_Mensal'] = df['Pagamento_Mensal'].fillna(df['Pagamento_Mensal'].median())
df['Total_Pago'] = df['Total_Pago'].fillna(df['Total_Pago'].median())

# Converter 'PhoneService' para float após substituição (opcional, se desejar numérico)
df['PhoneService'] = pd.to_numeric(df['PhoneService'], errors='coerce').astype('float')

# Verificar os dados faltantes após substituição
percentual_faltante = (df.isna().sum() / len(df)) * 100
print("Porcentagem de dados faltantes por coluna após substituição (%):")
print(percentual_faltante[percentual_faltante > 0].round(2))
print("\n")

# Exibir as primeiras linhas para confirmar
print("Primeiras linhas após substituição:")
df[['PhoneService', 'Pagamento_Mensal', 'Total_Pago']].head()

Porcentagem de dados faltantes por coluna após substituição (%):
Series([], dtype: float64)


Primeiras linhas após substituição:


PhoneService  Pagamento_Mensal  Total_Pago
0             0.0             29.85       29.85
1             0.1             56.95     1889.50
2             0.1             53.85      108.15
300           0.1             64.90     1509.80
301           0.1             95.45      396.10

Impacto: Substituir menos de 3% dos dados por esses métodos mantém a integridade da análise sem perda de linhas.

Alternativa: Exclusão (usada na etapa anterior) poderia ser preferida se os dados faltantes não fossem aleatórios, mas a substituição é válida aqui devido à baixa incidência.

Próximos Passos: Com os dados completos, você pode prosseguir com análises estatísticas ou modelagem.

**Dica variáveis categóricas:** Caso precise substituir uma variável categórica utilize um gráfico de barras para verificar a que mais aparece e utilizar ela na substituição.
Ou você pode optar por excluir a coluna, mas justifique sua escolha.

Caso opte por substituir você pode utilizar a função fillna:
df['SUACOLUNA'].fillna("VALOR QUE SUBSTITUIRA O NULO", inplace=True)
Ela funcionará como a replace.

# 3 - Verifique se encontramos valores digitados incorretamente, ou com letras maiusculas ou minusculas, ou até mesmo mesmos valores porém escritos de forma dirente e corrija.

In [622]:
df['Genero'] = df['Genero'].fillna('Unknown')

# Calcular a moda de PhoneService e lidar com caso vazio
phone_service_mode = df['PhoneService'].mode()
if not phone_service_mode.empty:
    df['PhoneService'] = df['PhoneService'].fillna(phone_service_mode[0])
else:
    df['PhoneService'] = df['PhoneService'].fillna('No')  # Valor padrão se moda não for calculada

# Converter PhoneService para float após substituição
df['PhoneService'] = pd.to_numeric(df['PhoneService'].replace({'Yes': 1.0, 'No': 0.0}), errors='coerce').astype('float')

df['Pagamento_Mensal'] = pd.to_numeric(df['Pagamento_Mensal'], errors='coerce').fillna(df['Pagamento_Mensal'].median())
df['Total_Pago'] = pd.to_numeric(df['Total_Pago'], errors='coerce').fillna(df['Total_Pago'].median())

# Verificar valores únicos após substituição
print("Valores únicos em PhoneService após substituição e conversão:")
print(df['PhoneService'].unique())
print("\n")

# Exibir as primeiras linhas para confirmar
print("Primeiras linhas após correção:")
print(df[['Genero', 'PhoneService', 'Pagamento_Mensal', 'Total_Pago']].head())#seu código aqui

Valores únicos em PhoneService após substituição e conversão:
[0.  0.1]


Primeiras linhas após correção:
     Genero  PhoneService  Pagamento_Mensal  Total_Pago
0    Female           0.0             29.85       29.85
1      Male           0.1             56.95     1889.50
2      Male           0.1             53.85      108.15
300    Male           0.1             64.90     1509.80
301  Female           0.1             95.45      396.10


Por algum motivo o PhoneService não tinha virado float, então transformei ele em float aqui mesmo

In [625]:
#seu código aqui
# Tratar strings vazias como NaN e substituir conforme etapa anterior
df = df.replace(r'^\s*$', pd.NA, regex=True)
df['PhoneService'] = df['PhoneService'].fillna(df['PhoneService'].mode()[0])
df['Pagamento_Mensal'] = pd.to_numeric(df['Pagamento_Mensal'], errors='coerce').fillna(df['Pagamento_Mensal'].median())
df['Total_Pago'] = pd.to_numeric(df['Total_Pago'], errors='coerce').fillna(df['Total_Pago'].median())

# Converter 'PhoneService' para float
df['PhoneService'] = pd.to_numeric(df['PhoneService'].replace({'Yes': 1.0, 'No': 0.0}), errors='coerce').astype('float')

# Ajustar a lista de colunas categóricas com base nas colunas reais
colunas_categoricas = [col for col in ['Genero', 'Casado', 'Dependents', 'PhoneService', 'Servico_Internet', 
                                      'Servico_Seguranca', 'Suporte_Tecnico', 'StreamingTV', 'Tipo_Contrato', 
                                      'PaymentMethod', 'Churn'] if col in df.columns]


# Função para padronizar e corrigir
def padronizar_coluna(coluna):
    if coluna in df.columns:
        # Aplicar .str.lower() apenas se a coluna for de texto (object ou category)
        if df[coluna].dtype in ['object', 'category']:
            df[coluna] = df[coluna].str.lower()
        
        # Correções específicas por coluna
        elif coluna == 'PhoneService':
            df[coluna] = df[coluna].replace({'yes': 1.0, 'no': 0.0}).astype('float')
        elif coluna in ['Casado', 'Dependents', 'Churn']:
            df[coluna] = df[coluna].replace({'yes': 'sim', 'no': 'nao'})
        elif coluna == 'Servico_Internet':
            df[coluna] = df[coluna].replace({'no internet service': 'no_internet_service'})
        elif coluna in ['Servico_Seguranca', 'Suporte_Tecnico', 'StreamingTV']:
            df[coluna] = df[coluna].replace({'no internet service': 'no_internet_service', 'yes': 'sim', 'no': 'nao'})
        elif coluna == 'Tipo_Contrato':
            df[coluna] = df[coluna].replace({
                'month-to-month': 'month_to_month',
                'one year': 'one_year',
                'two year': 'two_year'
            })
        elif coluna == 'PaymentMethod':
            df[coluna] = df[coluna].replace({
                'electronic check': 'electronic_check',
                'mailed check': 'mailed_check',
                'bank transfer': 'bank_transfer',
                'credit card': 'credit_card'
            })

# Aplicar padronização
for coluna in colunas_categoricas:
    padronizar_coluna(coluna)

# Exibir as primeiras linhas para confirmar
print("Primeiras linhas após padronização:")
df[colunas_categoricas].head()


Primeiras linhas após padronização:


Genero Casado Dependents  PhoneService Servico_Internet  \
0    female    yes         no           0.0              dsl   
1      male     no         no           0.1              dsl   
2      male     no         no           0.1              dsl   
300    male     no         no           0.1              dsl   
301  female     no         no           0.1      fiber optic   

    Servico_Seguranca Suporte_Tecnico StreamingTV   Tipo_Contrato  \
0                  no              no          no  month-to-month   
1                 yes              no          no        one year   
2                 yes              no          no  month-to-month   
300                no              no         yes        one year   
301                no              no         yes  month-to-month   

                 PaymentMethod Churn  
0             electronic check    no  
1                 mailed check    no  
2                 mailed check   yes  
300  bank transfer (automatic)    no  
301           electronic check   yes

# [EXTRA] 4 - Caso deseje, para manter um padrão na sua base, renomeie as colunas ou deixand

In [628]:
df.rename(columns={'PaymentMethod': 'Metodo_de_Pagamento'}, inplace=True)

In [630]:
df.head()

customerID  Genero  Idoso Casado Dependents  Tempo_como_Cliente  \
0    7590-VHVEG  female      0    yes         no                   1   
1    5575-GNVDE    male      0     no         no                  34   
2    3668-QPYBK    male      0     no         no                   2   
300  0895-LMRSF    male      0     no         no                  23   
301  8098-LLAZX  female      1     no         no                   4   

     PhoneService Servico_Internet Servico_Seguranca Suporte_Tecnico  \
0             0.0              dsl                no              no   
1             0.1              dsl               yes              no   
2             0.1              dsl               yes              no   
300           0.1              dsl                no              no   
301           0.1      fiber optic                no              no   

    StreamingTV   Tipo_Contrato        Metodo_de_Pagamento  Pagamento_Mensal  \
0            no  month-to-month           electronic check             29.85   
1            no        one year               mailed check             56.95   
2            no  month-to-month               mailed check             53.85   
300         yes        one year  bank transfer (automatic)             64.90   
301         yes  month-to-month           electronic check             95.45   

     Total_Pago Churn  
0         29.85    no  
1       1889.50    no  
2        108.15   yes  
300     1509.80    no  
301      396.10   yes

In [632]:
df.drop(columns=['customerID'], inplace=True)

In [634]:
df.head()


Genero  Idoso Casado Dependents  Tempo_como_Cliente  PhoneService  \
0    female      0    yes         no                   1           0.0   
1      male      0     no         no                  34           0.1   
2      male      0     no         no                   2           0.1   
300    male      0     no         no                  23           0.1   
301  female      1     no         no                   4           0.1   

    Servico_Internet Servico_Seguranca Suporte_Tecnico StreamingTV  \
0                dsl                no              no          no   
1                dsl               yes              no          no   
2                dsl               yes              no          no   
300              dsl                no              no         yes   
301      fiber optic                no              no         yes   

      Tipo_Contrato        Metodo_de_Pagamento  Pagamento_Mensal  Total_Pago  \
0    month-to-month           electronic check             29.85       29.85   
1          one year               mailed check             56.95     1889.50   
2    month-to-month               mailed check             53.85      108.15   
300        one year  bank transfer (automatic)             64.90     1509.80   
301  month-to-month           electronic check             95.45      396.10   

    Churn  
0      no  
1      no  
2     yes  
300    no  
301   yes